src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -236,156 +234,142 @@ public unsafe bool ParseHeaders(TRequestHandler handler, ref SequenceReader<byte
                         length = span.IndexOf(ByteLF) + 1;
                         if (length > 0)
                         {
-                            // Potentially found the end, or an invalid header.
-                            fixed (byte* pHeader = span)
+                            if (length < 5 ||
+                                span[(length - 2)] != ByteCR ||
+                                !TryTakeSingleHeader(handler, span[..(length - 2)])) // Do not include CRLF in span passed
```

Comment:
```suggestion
                                !TryTakeSingleHeader(handler, span[..^2])) // Do not include CRLF in span passed
```

Reply:
Wouldn't work, `length` is not `span.Length`